### ISSUE TITLE
Error running USDU with Flux using Anywhere Nodes

### ISSUE THREAD

**Issue by gonzalu** [ssitu/ComfyUI_UltimateSDUpscale#105]
Not sure if the problem is the use of the Anywhere nodes, but I keep getting this error:

```
Requested to load FluxClipModel_
Loading 1 new model
loaded completely 0.0 9319.23095703125 True
Canva size: 2080x3040
Image size: 832x1216
Scale factor: 3
Upscaling iteration 1 with scale factor 3
Tile size: 1040x1012
Tiles amount: 8
Grid: 4x2
Redraw enabled: True
Seams fix mode: NONE
!!! Exception during processing !!! too many values to unpack (expected 2)
Traceback (most recent call last):
  File "E:\ComfyUI\ComfyUI\execution.py", line 323, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\execution.py", line 198, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\execution.py", line 169, in _map_node_over_list
    process_inputs(input_dict, i)
  File "E:\ComfyUI\ComfyUI\execution.py", line 158, in process_inputs
    results.append(getattr(obj, func)(**inputs))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\nodes.py", line 151, in upscale
    processed = script.run(p=self.sdprocessing, _=None, tile_width=self.tile_width, tile_height=self.tile_height,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 565, in run
    upscaler.process()
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 138, in process
    self.image = self.redraw.start(self.p, self.image, self.rows, self.cols)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 245, in start
    return self.linear_process(p, image, rows, cols)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 180, in linear_process
    processed = processing.process_images(p)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\modules\processing.py", line 165, in process_images
    positive_cropped = crop_cond(p.positive, crop_region, p.init_size, init_image.size, tile_size)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\utils.py", line 452, in crop_cond
    for emb, x in cond:
        ^^^^^^
ValueError: too many values to unpack (expected 2)
```

**Comment by gonzalu** [comment]
![image](https://github.com/user-attachments/assets/8747c578-235d-4a8e-b3a1-44562e8bc7d3)

**Comment by gonzalu** [comment]
If I set the mode to type to NONE it works fine. Linear and Chess will not go through

**Comment by ssitu** [comment]
NONE just skips the resampling process, so it will just upscale using the upscale model. Does it work without anywhere nodes? It might just be an issue with your positive conditioning, which I don't see in the picture.

**Comment by gonzalu** [comment]
> NONE just skips the resampling process, so it will just upscale using the upscale model. Does it work without anywhere nodes? It might just be an issue with your positive conditioning, which I don't see in the picture.

Hmmm... let me take a look.

**Comment by gonzalu** [comment]
So, I connected it up without Anywhere nodes... and still got error...  :(

![image](https://github.com/user-attachments/assets/f7f86022-8b2f-47be-b8da-1b481e29bfc1)

![image](https://github.com/user-attachments/assets/843005ae-1f59-4b8f-a1d1-680a0446ac63)


```
!!! Exception during processing !!! too many values to unpack (expected 2)
Traceback (most recent call last):
  File "E:\ComfyUI\ComfyUI\execution.py", line 323, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\execution.py", line 198, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\execution.py", line 169, in _map_node_over_list
    process_inputs(input_dict, i)
  File "E:\ComfyUI\ComfyUI\execution.py", line 158, in process_inputs
    results.append(getattr(obj, func)(**inputs))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\nodes.py", line 151, in upscale
    processed = script.run(p=self.sdprocessing, _=None, tile_width=self.tile_width, tile_height=self.tile_height,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 565, in run
    upscaler.process()
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 138, in process
    self.image = self.redraw.start(self.p, self.image, self.rows, self.cols)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 245, in start
    return self.linear_process(p, image, rows, cols)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 180, in linear_process
    processed = processing.process_images(p)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\modules\processing.py", line 165, in process_images
    positive_cropped = crop_cond(p.positive, crop_region, p.init_size, init_image.size, tile_size)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale\utils.py", line 452, in crop_cond
    for emb, x in cond:
        ^^^^^^
ValueError: too many values to unpack (expected 2)
```

**Comment by gonzalu** [comment]
Just in case it matters, I am using the official Flux.1 Dev model file from BFL diffusers repo on huggingface, not schnell and not one of the other quantized versions floating around

**Comment by ssitu** [comment]
So it's a model issue? 1.5 models still seem to work, correct me if I'm wrong. I don't think I can run the new flux models with my 4 gb gpu, so this will be hard to fix.

**Comment by gonzalu** [comment]
So sorry... perhaps that's the problem. 

No worries and thank you so much for your help and for a great tool. I have used it for many many upscales :D 

Yes, I am using the full 23GB Flux model. 

Is your node using the same transformer and tensor code from the core comfyui nodes? Maybe that's why if they are different? 

Looking forward to when it works again. 

Cheers and Thank you!

**Comment by ssitu** [comment]
Hmmm you have a point. I can try fixing it if you are willing to test it for me. I can make a pull request with a potential fix for you or anyone else to clone and test.

**Comment by gonzalu** [comment]
Absolutely... More than happy to beta or alpha test :D 

I wish I could contribute but I am not a coder :(

**Comment by ssitu** [comment]
Alright, I'll have a go at it. If you want to help, I am assuming you have git installed.
If you are able to, open a terminal in the directory with this node (For example: C:/.../ComfyUI/custom_nodes/ComfyUI_UltimateSDUpscale)

Then execute the following command: `git checkout fix-crop-conditioning-loop`
This will make it so that you have my potential fix.
When you are done testing, then use this command to revert back to the main code: `git checkout main`

Try out my fix and let me know if it works with flux dev. It'd be nice if it works first try, but that might not happen.

**Comment by gonzalu** [comment]
Awesome.  Testing now

Manny Gonzalez

**Comment by gonzalu** [comment]
Oops... ran into trouble. 

I am using ComfyUI Portable. 

I get this error:

```
E:\ComfyUI\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale>git checkout fix-crop-conditioning-loop
error: pathspec 'fix-crop-conditioning-loop' did not match any file(s) known to git
```

**Comment by gonzalu** [comment]
...and I installed it via ComfyUI manager ;)

**Comment by gonzalu** [comment]
OK I think I found a workaround. I downloaded the branch and just copied it over the existing install ... not sure if that was correct... 

however, I got this error 

```
Canva size: 2688x1536
Image size: 1344x768
Scale factor: 2
Upscaling iteration 1 with scale factor 2
Tile size: 1024x1024
Tiles amount: 6
Grid: 2x3
Redraw enabled: True
Seams fix mode: NONE
!!! Exception during processing !!! 'str' object has no attribute 'copy'
Traceback (most recent call last):
  File "E:\ComfyUI\ComfyUI\execution.py", line 323, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\execution.py", line 198, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\execution.py", line 169, in _map_node_over_list
    process_inputs(input_dict, i)
  File "E:\ComfyUI\ComfyUI\execution.py", line 158, in process_inputs
    results.append(getattr(obj, func)(**inputs))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\USDU\nodes.py", line 151, in upscale
    processed = script.run(p=self.sdprocessing, _=None, tile_width=self.tile_width, tile_height=self.tile_height,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\USDU\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 565, in run
    upscaler.process()
  File "E:\ComfyUI\ComfyUI\custom_nodes\USDU\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 138, in process
    self.image = self.redraw.start(self.p, self.image, self.rows, self.cols)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\USDU\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 245, in start
    return self.linear_process(p, image, rows, cols)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\USDU\repositories\ultimate_sd_upscale\scripts\ultimate-upscale.py", line 180, in linear_process
    processed = processing.process_images(p)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\USDU\modules\processing.py", line 165, in process_images
    positive_cropped = crop_cond(p.positive, crop_region, p.init_size, init_image.size, tile_size)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ComfyUI\ComfyUI\custom_nodes\USDU\utils.py", line 454, in crop_cond
    cond_dict = x.copy()
                ^^^^^^
AttributeError: 'str' object has no attribute 'copy'
```

**Comment by gonzalu** [comment]
![image](https://github.com/user-attachments/assets/6d3770ad-35c8-4937-aae5-37c124f548e2)

**Comment by ssitu** [comment]
Thanks for the testing. Nice workaround, I thought you might have git installed but that works too. Strange error, I tried using the same code as some of the built-in nodes.

Could you try one of the following nodes? They should be already included in your comfy installation:

Conditioning (Combine)
Conditioning (Average)
Conditioning (Concat)

I haven't really used these before, but I think you can just take two text encodes, input them into one of these nodes, and take the output as the new positive conditioning. 
So something like this: `text encode1 + text encode2 -> conditioning merger node -> KSampler`
I want you to use a regular KSampler or something similar that isn't broken, instead of USDU. Please let me know if you get an error or not using flux dev, and a picture of the relevant part of the workflow like how you did in your last comment.

**Comment by gonzalu** [comment]
Oh sure... I'll also share my workflow file :D 

Currently, I use the diffusers model format, not the checkpoint (if it matters.)

I will create a fresh workflow just to make sure we have a clean simple workflow instead of my complex one which might be the problem. 

Give me a few minutes.

**Comment by gonzalu** [comment]
Here is my PRODUCTION workflow which I was trying to include USDU into. I normally use McBoaty Upscaler nodes (as I helped in the beta testing :D) but it is very slow... I use USDU when I want to generate an upscale a bit faster. 
[MG 069 McBoaty fp8 Smerge 2x Upscale Conditioning Switcher McBoaty Upscale Florence2 Inspire Prompts from File Matteo Nodes AI Random Prompts Ollama LoRA Flux.1 Dev workflow.json](https://github.com/user-attachments/files/17351180/MG.069.McBoaty.fp8.Smerge.2x.Upscale.Conditioning.Switcher.McBoaty.Upscale.Florence2.Inspire.Prompts.from.File.Matteo.Nodes.AI.Random.Prompts.Ollama.LoRA.Flux.1.Dev.workflow.json)

**Comment by ssitu** [comment]
I don't think it's an issue with your complex workflow. Other people have the same issue, notably #107

**Comment by gonzalu** [comment]
Guess what? NO ERRORS!!!! 

![image](https://github.com/user-attachments/assets/9b396b5c-39aa-4066-9763-ce829159c7b6)

**Comment by gonzalu** [comment]
Still part of the complex workflow and still using Anything Everywhere nodes 😮 

![image](https://github.com/user-attachments/assets/e4d845d3-8ee6-4d6a-a491-964aec223dfa)

**Comment by gonzalu** [comment]
and it finished without problems
![image](https://github.com/user-attachments/assets/e2b87852-79ec-4a5e-9147-b0581f032cef)

![image](https://github.com/user-attachments/assets/f6b3d39f-9266-4ccc-99e2-dc113bd10f95)

**Comment by ssitu** [comment]
I'm not sure what's happening. Could you try removing the conditioning combine node now and see if that gives an error?

**Comment by gonzalu** [comment]
sure

**Comment by gonzalu** [comment]
OK, if I connect directly from my Conditioning Switch, I get an error 

![image](https://github.com/user-attachments/assets/160dc85b-a222-4c55-a475-b72a7cac6b1f)


Let me try again connecting directly from the clip text encoder

**Comment by gonzalu** [comment]
Connecting directly from my primary Clip Text Encode node, I get error. 

BUT
BUT

I am NOT using the one from ComfyUI core... I am using the one from cubiq (Comfy Essentials Pack) 

that HAS to be the problem :D 

![image](https://github.com/user-attachments/assets/17bdb390-1f90-44b0-b7d6-199467f19584)


![image](https://github.com/user-attachments/assets/7734f6b2-3230-410e-8467-11c57b1527ed)

**Comment by gonzalu** [comment]
I think I found MY problem

ComfyUI Core Clip Text Encode works... 

![image](https://github.com/user-attachments/assets/5c9f7d4e-1f12-4d61-a0b6-0347b8a17bd5)

ComfyUI Essentials Custom node Clip Text Encode does NOT work

![image](https://github.com/user-attachments/assets/7344a183-ede0-434e-b7f7-b12a2eec7631)

**Comment by ssitu** [comment]
Well that's interesting. So there's no problem with my node?

**Comment by gonzalu** [comment]
I guess... but why would it be any different than the regular clip text encode? Doesn't the KSampler or any sampler have to have the same exact code for it to work with the models? 

Maybe I'll see if Matteo can look into why his node doesn't work with USDU :( 

Thanks.
